firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutorTest.java

```diff
@@ -0,0 +1,47 @@
+package com.example.firstapp;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.Until;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import java.io.IOException;
+
+public class CUJExecutorTest {
+    private static final String BASIC_SAMPLE_PACKAGE = "com.example.firstapp";
+    private static final int TIMEOUT = 6000;
+
+
+
+    //Basic clicking test
+    @Test
+    public void countToFive() throws Exception {
+        String[] preCUJ = {};
+        String[] postCUJ = {BASIC_SAMPLE_PACKAGE, "click;COUNT", "click;COUNT", "click;COUNT", "click;COUNT", "click;COUNT"};
```

Comment:
Should have the following structure:

1. parse CLI arguments into objects
2. pass objects into `executeCUJ()`

Reply:
Not sure I understand what you mean by this. The idea was for the script arguments to be the two string arrays, pre/postCUJ

Reply:
The CLI can take in strings, but it should be immediately parsed into Java objects, and passed accordingly.  You would likely have a generic Action class, and then ClickAction, ScrollAction, etc.

Reply:
Furthermore, should not make the first argument be a special case.  It's better to have something like: "start:packageName".  Or have a CLI argument just for the user to specify the packageName.

Reply:
Ahhh I see. To be clear, though, even if I replace the format of "packageName" with "start; packageName" or "click; packageName", it will still need to be handled as a special case, since this is not the same (in terms of UiAutomator calls) as clicking on a view or editing a textbox

Reply:
The difference is that the special case is handled by a special class, rather than being explicitly handled by the calling method.  It makes the code cleaner and generalizes the solution (improving flexibility, readability, and testability).